sessionctx/variable/session.go

```diff
@@ -186,3 +192,40 @@ func (s *SessionVars) GetNextPreparedStmtID() uint32 {
 func (s *SessionVars) SetCurrentUser(user string) {
 	s.User = user
 }
+
+// SetSystemVar sets a system variable.
+func (s *SessionVars) SetSystemVar(key string, value types.Datum) error {
+	key = strings.ToLower(key)
+	var sVal string
+	var err error
+	if value.Kind() == types.KindNull {
+		delete(s.systems, key)
+		return nil
+	}
+	sVal, err = value.ToString()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	key = strings.ToLower(key)
```

Comment:
Just in case the key is not lower.
This API is not used frequently, the cost of `ToLower` is negligible.